### DISCUSSION TITLE
Library to use RF 433 modules

### DISCUSSION THREAD

**Discussion by ghost** [micropython/micropython#13284] [Show and tell]
Hey, I recentlt had to use the 433 modules and I ended up creating a small library that handles the protocol to send/recieve messages. Maybe it is usefull for someone else -> [https://github.com/knemay/rf433](https://github.com/knemay/rf433)

**Comment by peterhinch** [comment]
I hope you don't mind me chucking in a couple of comments but this piece of code looks unusual for two reasons:
```python
        while n < len(msg):
            new_time = time.ticks_us()
            if new_time < old_time:  # Handle overflows
                old_time = old_time - 1_073_741_824  # microseconds overflow = 2^30
            if new_time > old_time + self.baud_rate:
                self.trans.value(msg[n])
                n += 1
                old_time = new_time
```
1 .The recommended way to handle overflow is with [ticks_diff](https://docs.micropython.org/en/latest/library/time.html#time.ticks_diff).
2. The fragment `old_time + self.baud_rate` adds a time in μs to a frequency in Hz, which can't possibly be right :)

**Reply by ghost** [reply]
Appreciate the comment, I just updated it to use ticks_diff. And for the buad_rate thing, I was (wrongly) using that argument as period instead of baudrate, I just changed it to get the period using the baud rate so now the argument is actually the baudrate.
Any other comment is welcome,

PD: I'm a big fan of your work in micropython, I have learned a lot by reading your code and your guides, so thank you for everything
